hooks/RPM.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2016 Signal K <info@signalk.org> and contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+ const utils = require('nmea0183-utilities')
+
+/*
+#        0 1 2   3   4 5
+#        | | |   |   | |
+# $--RPM,a,x,x.x,x.x,A*hh<CR><LF> Field Number:
+#  0) Source, S = Shaft, E = Engine 1) Engine or shaft number 2) Speed,
+#  Revolutions per minute 3) Propeller pitch, % of maximum, "-" means
+#  astern 4) Status, A means data is valid 5) Checksum
+*/
+
+module.exports = function (parser, input) {
+  const { id, sentence, parts, tags } = input
+
+  try {
+    const delta = {
+      context: 'vessels.self',
+      updates: [
+        {
+          source: tags.source,
+          timestamp: tags.timestamp,
+          values: [
+            {
+              path: 'propulsion.engine1.revolutions',
```

Comment:
Not sure we can just use `engine1`.. it's an issue, as parser would have to know what engine the value would be coming from. One possible solution would be to make the context configurable - both at a global level (`new Parser({ context: 'vessels.foobar' })`) and a individual sentence level (`parser.parse(sentence, 'vessels.self.propulsion.engine1')`). In the latter case, the hook would need to be smart enough to recognise the context change the delta. 

I'll create a separate issue about this, and we can solve this later.